slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/InterchangeImportConflicts.java

```diff
@@ -0,0 +1,159 @@
+              .isEqualTo(
+                  Optional.of(
+                      Bytes.fromHexString(
+                          "0x4ff6f743a43f3b4f95350831aeaf0a122a1a392922c45d804280284a69eb850b")));
+        });
+  }
+
+  @Test
+  void doNotDuplicateAttestations() throws IOException {
+    final URL importFile = Resources.getResource("interchange/singleValidAttestation.json");
+    slashingProtection.importData(importFile.openStream());
+    slashingProtection.importData(importFile.openStream()); // attempt to reimport
+    jdbi.useHandle(
+        handle -> {
+          final List<SignedAttestation> attestationsInDb = findAllAttestations(handle);
+          assertThat(attestationsInDb).hasSize(1);
+          assertThat(attestationsInDb.get(0).getSourceEpoch()).isEqualTo(UInt64.valueOf(5));
+          assertThat(attestationsInDb.get(0).getTargetEpoch()).isEqualTo(UInt64.valueOf(6));
+          assertThat(attestationsInDb.get(0).getValidatorId()).isEqualTo(1);
+          assertThat(attestationsInDb.get(0).getSigningRoot())
+              .isEqualTo(Optional.of(Bytes.fromHexString("0x123456")));
+        });
+  }
+
+  @Test
+  void canLoadAFileWithDuplicateAttesationsButOnlyOneInserted() throws IOException {
+    final URL importFile = Resources.getResource("interchange/duplicateAttestation.json");
+    slashingProtection.importData(importFile.openStream());
+    slashingProtection.importData(importFile.openStream()); // attempt to reimport
+    jdbi.useHandle(
+        handle -> {
+          final List<SignedAttestation> attestationsInDb = findAllAttestations(handle);
+          assertThat(attestationsInDb).hasSize(1);
+          assertThat(attestationsInDb.get(0).getSourceEpoch()).isEqualTo(UInt64.valueOf(5));
+          assertThat(attestationsInDb.get(0).getTargetEpoch()).isEqualTo(UInt64.valueOf(6));
+          assertThat(attestationsInDb.get(0).getValidatorId()).isEqualTo(1);
+          assertThat(attestationsInDb.get(0).getSigningRoot())
+              .isEqualTo(Optional.of(Bytes.fromHexString("0x123456")));
+        });
+  }
+
+  @Test
+  void canLoadInterchangeFormatWithMissingSigningRootForBlock() throws IOException {
+    final URL importFile = Resources.getResource("interchange/singleNullSigningRootBlock.json");
+    slashingProtection.importData(importFile.openStream());
+    jdbi.useHandle(
+        handle -> {
+          final List<SignedBlock> blocksInDb = findAllBlocks(handle);
+          assertThat(blocksInDb).hasSize(1);
+          assertThat(blocksInDb.get(0).getSlot()).isEqualTo(UInt64.valueOf(12345));
+          assertThat(blocksInDb.get(0).getValidatorId()).isEqualTo(1);
+          assertThat(blocksInDb.get(0).getSigningRoot()).isEmpty();
+        });
+  }
+
+  @Test
+  void canLoadInterchangFormatWithMissingSigningRootForAttestation() throws IOException {
+    final URL importFile =
+        Resources.getResource("interchange/singleNullSigningROotAttestation.json");
```

Comment:
Think the filename should be "singleNullSigningRootAttestation". Not sure why it's not failing though.